### PR TITLE
Provide default platform_name everywhere

### LINKED ISSUE #325: Unmounted NRF52_DK shows error when `-u` is passed
Trying to list all devices yields.
```
mbedls --json                                                                    15:22:32   1 
WARNING:mbedls.lstools_base:MBED with target id '000682416648' is connected, but not mounted. Use the '-u' flag to include it in the list.
[]
```
Following the suggestion causes this error:
```
mbedls --json --list-unmounted                                                   15:22:20   0 
Traceback (most recent call last):
  File "/home/bernhard/.local/bin//mbedls", line 11, in <module>
    load_entry_point('mbed-ls==1.4.2', 'console_scripts', 'mbedls')()
  File "/home/bernhard/.local/lib/python3.6/site-packages/mbed_lstools/main.py", line 262, in mbedls_main
    ret_code = args.command(mbeds, args)
  File "/home/bernhard/.local/lib/python3.6/site-packages/mbed_lstools/main.py", line 138, in mbeds_as_json
    read_details_txt=True),
  File "/home/bernhard/.local/lib/python3.6/site-packages/mbed_lstools/lstools_base.py", line 150, in list_mbeds
    name = device['platform_name']
KeyError: 'platform_name'
```

### PR DESCRIPTION
**Pull request by theotherjimmy**
Resolves #325

### CODE REVIEW

**Comment by theotherjimmy** [comment]
@bridadan I think I got this to where I wanted. Could you review?

File: mbed_lstools/lstools_base.py [hunk 1]
```diff
@@ -140,6 +140,8 @@ def list_mbeds(
                         "Use the '-u' flag to include it in the list.",
                         device['target_id_usb_id'])
             else:
+                platform_data = self.plat_db.get(device['target_id_usb_id'][0:4], verbose_data=True)
+                device.update(platform_data or {"platform_name": None})
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: 💯

File: mbed_lstools/lstools_base.py [hunk 2]
```diff
@@ -167,9 +169,6 @@ def _fs_never(self, device, filter_function, read_details_txt):
         """Filter device without touching the file system of the device"""
         device['target_id'] = device['target_id_usb_id']
         device['target_id_mbed_htm'] = None
-        platform_data = self.plat_db.get(device['target_id'][0:4],
-                                         verbose_data=True)
-        device.update(platform_data or {'platform_name': None})
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: :wave:

File: mbed_lstools/lstools_base.py [hunk 3]
```diff
@@ -193,8 +192,6 @@ def _fs_after_id_check(self, device, filter_function, read_details_txt):
         """
         device['target_id'] = device['target_id_usb_id']
         device['target_id_mbed_htm'] = None
-        platform_data = self.plat_db.get(device['target_id'][0:4], verbose_data=True)
-        device.update(platform_data or {'platform_name': None})
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: 🍃

File: test/mbedls_toolsbase.py [hunk 4]
```diff
@@ -246,28 +247,34 @@ def test_fs_never(self):
...
+        self.base.list_unmounted = True
         with patch("mbed_lstools.lstools_base.MbedLsToolsBase._update_device_from_fs") as _up_fs:
             filter = None
-            ret = self.base._fs_never(deepcopy(device), filter, False)
-            ret_with_details = self.base._fs_never(deepcopy(device), filter, True)
-            self.assertIsNotNone(ret)
-            self.assertIsNotNone(ret_with_details)
-            self.assertEqual(ret['target_id'], ret['target_id_usb_id'])
-            self.assertEqual(ret, ret_with_details)
+            ret = self.base.list_mbeds(FSInteraction.Never, filter, read_details_txt=False)
```
Comment (bridadan) [inline test/mbedls_toolsbase.py]: Let's change the other tests `test_fs_after` and `test_fs_before` to go through `list_mbeds` as well so the cases are consistent.

File: test/mbedls_toolsbase.py [hunk 5]
```diff
@@ -351,6 +351,7 @@ def test_fs_before(self):
             'mount_point': 'invalid_mount_point',
             'serial_port': 'invalid_serial_port'
         }
+        self.base.list_unmounted = True
```
Comment (bridadan) [inline test/mbedls_toolsbase.py]: Pretty sure I'm not getting this cuz my brain is dead, but why is this needed?

Comment (theotherjimmy) [reply test/mbedls_toolsbase.py]: Cuz the test mount points don't exist.

Comment (bridadan) [reply test/mbedls_toolsbase.py]: Summing up what we discussed offline: we should be mocking anytime the filesystem would be accessed. So in this case we need to mock the correct `find_candidates` call.
